### ISSUE TITLE
f strings with `=` don't get formatted

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#11005]
# before
```py
a = 1
f"{ a   =  }"
f"{ a   }"
```
# after
```py
a = 1
f"{ a   =  }"
f"{a}"
```

[playground](https://play.ruff.rs/ee8c1c05-db9e-41e4-8ad6-7dc96248ebb7)

**Comment by charliermarsh** [comment]
I think there is a reason for this, maybe that those are debug representations and they must be verbatim? @dhruvmanila would know.

**Comment by DetachHead** [comment]
oh i didn't realize the whitespace is included in the output
```
>>> f"{1   =  }"  
'1   =  1'
```

in that case, maybe it can be a lint rule instead so people can decide for themselves whether they want to enforce a consistent formatting for these

**Comment by dhruvmanila** [comment]
Yeah, so as you've noticed whitespaces are preserved for debug expressions which means if the formatter removed them, it will actually change the code output. This is why we avoid doing that.

I don't know if it can be a lint rule as we can't infer if the user intended to put the whitespaces there or not.

**Comment by dhruvmanila** [comment]
I'll close this issue as it's working as expected.
